src/search/tasks/negated_axioms_task.cc

```diff
@@ -0,0 +1,358 @@
+        return;
+    }
+
+    const set<FactPair> &conditions = conditions_as_cnf[index];
+    for (const FactPair &fact_pair : conditions) {
+        // The current set is covered with the current hitting set elements.
+        if (hitting_set.find(fact_pair) != hitting_set.end()) {
+            collect_non_dominated_hitting_sets_recursively(
+                conditions_as_cnf, index + 1, hitting_set, results);
+            return;
+        }
+    }
+
+    for (const FactPair &elem : conditions) {
+        // We don't allow choosing different facts from the same variable.
+        bool same_var_occurs_already = false;
+        for (const FactPair &f : hitting_set) {
+            if (f.var == elem.var) {
+                same_var_occurs_already = true;
+                break;
+            }
+        }
+        if (same_var_occurs_already) {
+            continue;
+        }
+
+        set<FactPair> chosen_new(hitting_set);
+        chosen_new.insert(elem);
+        collect_non_dominated_hitting_sets_recursively(
+            conditions_as_cnf, index + 1, chosen_new, results);
+    }
+}
+
+
+int NegatedAxiomsTask::get_operator_cost(int index, bool is_axiom) const {
+    if (!is_axiom || index < negated_axioms_start_index) {
+        return parent->get_operator_cost(index, is_axiom);
+    }
+
+    return 0;
+}
+
+string NegatedAxiomsTask::get_operator_name(int index, bool is_axiom) const {
+    if (!is_axiom || index < negated_axioms_start_index) {
+        return parent->get_operator_name(index, is_axiom);
+    }
+
+    return "<axiom>";
+}
+
+int NegatedAxiomsTask::get_num_operator_preconditions(int index, bool is_axiom) const {
+    if (!is_axiom || index < negated_axioms_start_index) {
+        return parent->get_num_operator_preconditions(index, is_axiom);
+    }
+
+    return negated_axioms[index - negated_axioms_start_index].condition.size();
+}
+
+FactPair NegatedAxiomsTask::get_operator_precondition(
+    int op_index, int fact_index, bool is_axiom) const {
+    if (!is_axiom || (op_index < negated_axioms_start_index)) {
+        return parent->get_operator_precondition(op_index, fact_index, is_axiom);
+    }
+    return negated_axioms[op_index - negated_axioms_start_index].condition[fact_index];
+}
+
+int NegatedAxiomsTask::get_num_operator_effects(int op_index, bool is_axiom) const {
+    if (!is_axiom || op_index < negated_axioms_start_index) {
+        return parent->get_num_operator_effects(op_index, is_axiom);
+    }
+
+    return 1;
+}
+
+int NegatedAxiomsTask::get_num_operator_effect_conditions(
+    int op_index, int eff_index, bool is_axiom) const {
+    if (!is_axiom || op_index < negated_axioms_start_index) {
+        return parent->get_num_operator_effect_conditions(op_index, eff_index, is_axiom);
+    }
+
+    return 0;
+}
+
+FactPair NegatedAxiomsTask::get_operator_effect_condition(
+    int op_index, int eff_index, int cond_index, bool is_axiom) const {
+    assert(!is_axiom || op_index < negated_axioms_start_index);
```

Comment:
The handling of NegatedAxioms and Axioms is not consistent in this transformation. If I am not mistaken, the body of a normal axiom is treated as an effect condition, whereas you treat the ones of a negated axiom as precondition. Right?

Reply:
I adjusted the functions accordingly. As far as I can tell, axioms still have one precondition, namely the value of the derived variable before the axiom is applied. I specified this as `1 - head.value`, (where `head.value` is the value the derived variable will change to). This only works for binary derived variables, but we have this assumption pretty much everywhere in the code anyways.